ParseUI/Classes/QueryCollectionViewController/PFQueryCollectionViewController.h

```diff
@@ -0,0 +1,201 @@
+ */
+@property (nonatomic, assign) BOOL paginationEnabled;
+
+/*!
+ @abstract The number of objects to show per page. Default - `25`.
+ */
+@property (nonatomic, assign) NSUInteger objectsPerPage;
+
+/*!
+ @abstract Whether the collection is actively loading new data from the server.
+ */
+@property (nonatomic, assign, getter=isLoading) BOOL loading;
+
+///--------------------------------------
+/// @name Creating a PFQueryCollectionViewController
+///--------------------------------------
+
+/*!
+ @abstract Initializes a view controller with a `UICollectionViewFlowLayout` and a class name
+ of <PFObject> that will be associated with this collection.
+
+ @param className The class name of the instances of <PFObject> that this table will display.
+
+ @returns An initialized `PFQueryCollectionViewController` object or `nil` if the object couldn't be created.
+ */
+- (instancetype)initWithClassName:(NSString *)className;
+
+/*!
+ @abstract Initializes a view controller with a class name of <PFObject> that will be associated with this collection.
+
+ @param layout    Layout for collection view to use.
+ @param className The class name of the instances of <PFObject> that this table will display.
+
+ @returns An initialized `PFQueryCollectionViewController` object or `nil` if the object couldn't be created.
+ */
+- (instancetype)initWithCollectionViewLayout:(UICollectionViewLayout *)layout className:(NSString *)className NS_DESIGNATED_INITIALIZER;
+
+///--------------------------------------
+/// @name Responding to Events
+///--------------------------------------
+
+/*!
+ Called when objects will loaded from Parse. If you override this method, you must
+ call [super objectsWillLoad] in your implementation.
+ */
+- (void)objectsWillLoad NS_REQUIRES_SUPER;
+
+/*!
+ Called when objects have loaded from Parse. If you override this method, you must
+ call [super objectsDidLoad:] in your implementation.
+ @param error The Parse error from running the PFQuery, if there was any.
+ */
+- (void)objectsDidLoad:(NSError *)error NS_REQUIRES_SUPER;
+
+///--------------------------------------
+/// @name Accessing Results
+///--------------------------------------
+
+/*!
+ @abstract The array of instances of <PFObject> that is used as a data source.
+ */
+@property (nonatomic, copy, readonly) NSArray *objects;
+
+/*!
+ @abstract Returns an object at a particular indexPath.
+
+ @discussion The default impementation returns the object at `indexPath.item`.
+ If you want to return objects in a different indexPath order, like for sections, override this method.
+
+ @param indexPath An instance of `NSIndexPath`.
+
+ @returns The object at the specified indexPath.
+ */
+- (PFObject *)objectAtIndexPath:(NSIndexPath *)indexPath;
+
+/*!
+ @abstract Clears the collection view of all objects.
+ */
+- (void)clear;
+
+/*!
+ @abstract Clears the collection view and loads the first page of objects.
+ */
+- (void)loadObjects;
```

Comment:
hmm. I feel that `refreshObjects` is more suitable in this case since we're clearing the collection. But `loadObjects` is fine, too